quickwit-doc-mapper/src/tokenizers.rs

```diff
@@ -17,34 +17,399 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+use std::str::CharIndices;
+
 use once_cell::sync::Lazy;
-use tantivy::tokenizer::{RawTokenizer, RemoveLongFilter, TextAnalyzer, TokenizerManager};
+use regex::Regex;
+use tantivy::tokenizer::{
+    BoxTokenStream, RawTokenizer, RemoveLongFilter, TextAnalyzer, Token, TokenStream, Tokenizer,
+    TokenizerManager,
+};
+
+static REGEX_ERROR_MSG: &str = "Failed to compile regular expression. This should never happen! Please, report on https://github.com/quickwit-oss/quickwit/issues.";
+
+// The array is ordered according to how often we'd stumble upon the expression.
+// If an expression is bound to appear a lot, it should be at the smallest index possible
+// to avoid iterating through the array as much as possible.
+static REGEX_ARRAY: Lazy<[Regex; 4]> = Lazy::new(|| {
+    [
+        // Regex to match numbers or numbers with certain characters that should not be split on
+        // (e.g IP addresses)
+        Regex::new("^[0-9][-/%_\\.:a-zA-Z0-9]*").expect(REGEX_ERROR_MSG),
+        // Date regex to match dates similar to "MMM d" and potentially followed by "yyyy" or a
+        // time format.
+        Regex::new("^[A-Za-z]{1,3}[ \t]{1,2}[0-9]{1,2}([ \t]{1,2}[0-9]+[-_/:0-9]*)?")
+            .expect(REGEX_ERROR_MSG),
+        // URL regex to match http, https or relative URL
+        // The regex is build to match:
+        //      Protocol followed by a domain name and domain extension and finally
+        //      the path and query if any
+        Regex::new("^(https?://)?[a-z0-9][-\\.a-z0-9]*\\.[\\.a-z]+(/[-&\\?a-z.=]*)*")
+            .expect(REGEX_ERROR_MSG),
+        // Regex to match unix filesystem paths or endpoints of an HTTP request.
+        Regex::new("^/?[-a-zA-Z0-9./:%_]+[a-zA-Z0-9]+/?").expect(REGEX_ERROR_MSG),
+    ]
+});
+
+/// Log friendly Tokenizer that avoids splittings on ponctuation in:
```

Comment:
Globally this is true but.... I was a bit surprised when I saw these 2 things:
- "2017-05-16_13:55:31" is not watched by the datetime regex but by the first regex "^[0-9][-/%_\.:a-zA-Z0-9]*"
- "req-b6a4fa91-7414-432a-b725-52b5613d3ca3" is matched by the URL regex "^/?[-a-zA-Z0-9./:%_]+[a-zA-Z0-9]+/?"

There is also a URL regex and UNIX filesystem regex, this seems redundant.

My global feeling is that it works, but it's messy.